CMakeLists.txt

```diff
@@ -59,42 +71,46 @@ if (NOT (CMAKE_USE_PTHREADS_INIT OR CMAKE_USE_WIN32_THREADS_INIT) OR EMSCRIPTEN
 endif()
 
 # Customize the build by passing "-D<option_name>=ON|OFF" in the command line.
-option(BUILD_SHARED_LIBS "Build shared libraries" ON)
```

Comment:
This option was particularly problematic because it's actually [a standard cmake variable](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html) so declaring it as an option here forces it ON globally for the whole project. This means that when bdwgc is included as a cmake subdir, targets higher up the tree will start unexpectedly building as dynamic libraries.